### DISCUSSION TITLE
bug : GGUF model file upload not working

### DISCUSSION THREAD

**Discussion by hemangjoshi37a** [open-webui/open-webui#2805] [General]
GGFU model file upload not working

**Comment by ihor-sokoliuk** [comment]
I will add more info there as I am facing it as well.
When I select a GGUF file for upload and click "upload", the UI resets like I did not choose anything, and that's it. There is no error message on the UI.

In requests, there is one post request: 
https://ollama/api/v1/utils/upload

With response:
413 Request Entity Too Large

The file I tried to upload is over 40 gigs, taken from there: 
https://huggingface.co/senseable/MoMo-70B-lora-1.8.6-DPO-gguf/tree/main

There is probably a config that adjusts the maximum file size to upload.

Also, it will be great to have the option to download GUFFs by URL directly into Ollama.

I hope it helps!

**Comment by justinh-rahb** [comment]
@ihor-sokoliuk, GGUF download by URL is already possible, just click the `file mode` to toggle it to `URL mode`:

<img width="445" alt="Screenshot 2024-01-17 at 12 05 37 PM" src="https://github.com/ollama-webui/ollama-webui/assets/52832301/30d5b1e9-627d-4fdb-a9e3-6cb71f15be5c">

**Comment by ihor-sokoliuk** [comment]
It saved me today! Thank you @justinh-rahb 
Then, only the upload feature requires attention.

**Comment by tjbck** [comment]
I'm aware that there is somewhat of an issue with indicating the upload progress, but besides the progress bar, everything else should work as intended (tested with uploading 2gb model)!

**Comment by hemangjoshi37a** [comment]
Should I upload only the zip  file or all the files in a gguf model repo ?

```
singletons.17816556.js:1 [Intervention] Slow network is detected. See https://www.chromestatus.com/feature/5636954674692096 for more details. Fallback font will be used while loading: chrome-extension://liecbddmkiiihnedobmlmillhodjkdmb/fonts/CircularXXWeb-Book.woff2
0.d1b35dfe.js:1 Object
2.08256b6d.js:52 IDB Not Found
2.08256b6d.js:52 0.1.20
2.08256b6d.js:35 settings true
/openai/api/models:1 
        
        
       Failed to load resource: the server responded with a status of 401 ()
index.fe45cfdc.js:1 Object
2.08256b6d.js:52 OpenAI: Network Problem
5.3e6df63a.js:1 Object
2.08256b6d.js:35 Object
5.3e6df63a.js:1 Object
3.f588ab9c.js:1 initNewChat
3.f588ab9c.js:1 
3.f588ab9c.js:1 Object
3.f588ab9c.js:1 submitPrompt 
3.f588ab9c.js:1 Array(0)
3.f588ab9c.js:2 dolphin-mixtral:8x7b-v2.7-q2_K
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:09.465545924Z","message":{"role":"assistant","content":" Hello"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:09.936658327Z","message":{"role":"assistant","content":"!"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:10.389682679Z","message":{"role":"assistant","content":" How"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:11.053123898Z","message":{"role":"assistant","content":" can"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:11.520292859Z","message":{"role":"assistant","content":" I"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:11.9377433Z","message":{"role":"assistant","content":" assist"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:12.339885641Z","message":{"role":"assistant","content":" you"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:12.966958336Z","message":{"role":"assistant","content":" today"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:13.608430777Z","message":{"role":"assistant","content":"?"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:14.035827598Z","message":{"role":"assistant","content":" Please"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:14.616952962Z","message":{"role":"assistant","content":" provide"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:15.089691725Z","message":{"role":"assistant","content":" more"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:15.615575237Z","message":{"role":"assistant","content":" information"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:16.16073538Z","message":{"role":"assistant","content":" about"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:16.558060174Z","message":{"role":"assistant","content":" your"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:17.08179965Z","message":{"role":"assistant","content":" request"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:17.658254795Z","message":{"role":"assistant","content":" so"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:18.258579865Z","message":{"role":"assistant","content":" I"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:18.849906442Z","message":{"role":"assistant","content":" can"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:19.41702565Z","message":{"role":"assistant","content":" help"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:19.9127692Z","message":{"role":"assistant","content":" you"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:20.394066877Z","message":{"role":"assistant","content":" better"},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:20.815378983Z","message":{"role":"assistant","content":"."},"done":false}
3.f588ab9c.js:4 {"model":"dolphin-mixtral:8x7b-v2.7-q2_K","created_at":"2024-01-19T03:55:21.387473796Z","message":{"role":"assistant","content":""},"done":true,"total_duration":77946268086,"load_duration":48780447426,"prompt_eval_count":31,"prompt_eval_duration":17698322000,"eval_count":23,"eval_duration":11450638000}
2.08256b6d.js:35 FileList0: FilelastModified: 1705487847979lastModifiedDate: Wed Jan 17 2024 16:07:27 GMT+0530 (India Standard Time)[[Prototype]]: Objectname: "mpt-7b-instruct-Q4_0.gguf"size: 3912127328type: ""webkitRelativePath: ""[[Prototype]]: FilelastModified: (...)lastModifiedDate: (...)name: (...)webkitRelativePath: (...)constructor: ƒ File()arguments: nullcaller: nulllength: 2name: "File"prototype: File {constructor: ƒ, …}[[Prototype]]: ƒ Blob()[[Scopes]]: Scopes[0]Symbol(Symbol.toStringTag): "File"size: (...)type: (...)get lastModified: ƒ lastModified()length: 0name: "get lastModified"arguments: (...)caller: (...)[[Prototype]]: ƒ ()apply: ƒ apply()arguments: (...)bind: ƒ bind()call: ƒ call()caller: (...)constructor: ƒ Function()length: 0name: ""toString: ƒ toString()Symbol(Symbol.hasInstance): ƒ [Symbol.hasInstance]()get arguments: ƒ ()set arguments: ƒ ()get caller: ƒ ()set caller: ƒ ()[[FunctionLocation]]: [[Prototype]]: Object[[Scopes]]: Scopes[0][[Scopes]]: Scopes[0]No propertiesget lastModifiedDate: ƒ lastModifiedDate()length: 0name: "get lastModifiedDate"arguments: (...)caller: (...)[[Prototype]]: ƒ ()[[Scopes]]: Scopes[0]get name: ƒ name()length: 0name: "get name"arguments: (...)caller: (...)[[Prototype]]: ƒ ()[[Scopes]]: Scopes[0]get webkitRelativePath: ƒ webkitRelativePath()length: 0name: "get webkitRelativePath"arguments: (...)caller: (...)[[Prototype]]: ƒ ()[[Scopes]]: Scopes[0][[Prototype]]: BlobarrayBuffer: ƒ arrayBuffer()size: (...)slice: ƒ slice()stream: ƒ stream()text: ƒ text()type: (...)constructor: ƒ Blob()Symbol(Symbol.toStringTag): "Blob"get size: ƒ size()get type: ƒ type()[[Prototype]]: Objectlength: 1[[Prototype]]: FileListitem: ƒ item()length: (...)constructor: ƒ FileList()Symbol(Symbol.iterator): ƒ values()Symbol(Symbol.toStringTag): "FileList"get length: ƒ length()[[Prototype]]: Object
/api/v1/utils/upload:1 
        
        
       Failed to load resource: net::ERR_CONNECTION_RESET
2.08256b6d.js:30 TypeError: Failed to fetch
    at window.fetch (start.e5e94b7c.js:1:1410)
    at ne (2.08256b6d.js:30:1170)
    at HTMLFormElement.rs (2.08256b6d.js:35:3847)
    at HTMLFormElement.<anonymous> (scheduler.4bc66f67.js:1:3673)
start.e5e94b7c.js:1 Fetch finished loading: GET "https://ollama.hjlabs.in/ollama/api/tags".
window.fetch @ start.e5e94b7c.js:1
u @ index.d7d99b66.js:1
Ne @ 2.08256b6d.js:35
ne @ 2.08256b6d.js:35
```

here is chrome dev console log

**Comment by tjbck** [comment]
Please provide us with the steps to reproduce, thanks.

**Comment by hemangjoshi37a** [comment]
1 : download file `[mpt-7b-instruct-Q4_0.gguf`  from  [https://huggingface.co/filipealmeida/mpt-7b-instruct-GGUF/blob/main/mpt-7b-instruct-Q4_0.gguf](https://huggingface.co/filipealmeida/mpt-7b-instruct-GGUF/blob/main/mpt-7b-instruct-Q4_0.gguf)
2 : try to upload that file using GGUF file upload

**Comment by hemangjoshi37a** [comment]
![Screenshot from 2024-01-19 11-23-41](https://github.com/ollama-webui/ollama-webui/assets/12392345/b8cf3330-f1ea-4585-84a0-fa1f4ded9019)

**Comment by bmabir17** [comment]
I am also facing this issue, after selecting the file and pressing upload button. The UI shows a upload loader with 0%. Then in the console it shows 
```
 SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data index.2dec9879.js:1:1887

OpenAI: Network Problem 2.1a0e1f1a.js:33:755

TypeError: NetworkError when attempting to fetch resource. 
```



My network do not have access to openAI api, And i do not intend to use it. What can be the solution for this?

**Comment by mhussaincov94** [comment]
I also am faceing this issue.
no upload progress is displayed.
I have moddles localy is there any other way to import them untill a fix is added?
I would be greatful for any help.
Majid

**Comment by KevinKrueger** [comment]
Where are the uploaded models located?
Can I put them in the directory for the first time?

**Comment by zer0ish** [comment]
I'm using Unraid.
Using the downloaded gguf file, doesn't seem to work. 
It's stuck at 0%. Looking at my ollama share, there isn't any new file being uploaded anywhere. 
Edit: I let it run all night, when I looked at it, the process seemed to have finished but nothing new in my usable models.
![image](https://github.com/open-webui/open-webui/assets/29928635/37446263-6382-43a3-b634-b119ea2b7d72)

But using the link from huggingface somewhat works? 
Link used: https://huggingface.co/cjpais/llava-1.6-mistral-7b-gguf/resolve/main/llava-v1.6-mistral-7b.Q8_0.gguf?download=true

The progress bar moves up to 100%, I see the sha256 file in my docker share for ollama, it has the proper size for the model I'm trying to get, but it just stays on a rotating circle indicating it's doing something. But it's been doing something for an hour.

You can see the circle progress like it's trying to do something bellow the "URL Mode" text.
![image](https://github.com/open-webui/open-webui/assets/29928635/521d4972-cdf7-4568-964e-b1779ecda1be)

All the regular models from ollama.com work fine for me including Llava model which was broken until the ollama 0.1.29 update.
I know it's in Experimental, so I'm not to concerned about it yet.

**Comment by imadreamerboy** [comment]
I have the same error, i get this output:
![grafik](https://github.com/open-webui/open-webui/assets/48453988/13e440cd-7b78-4f80-ab9d-8f2347cbcd8b)
Ollama log:
[GIN] 2024/03/25 - 00:18:24 | 404 |            0s |       127.0.0.1 | POST     "/blobs/sha256:0068f25d1fc37cb25aa6be85064432eeeb1a0754d97139c0d2eb3529fc8fc32b"

**Comment by syberphunk** [comment]
I'm experiencing this problem also, except it's saying that the connection was aborted when attempting to upload. The gguf file appears in the upload folder but doesn't get any further

![image](https://github.com/open-webui/open-webui/assets/234857/caaffede-be27-45cd-a02c-587fc8fbaeaf)

**Reply by iYonga** [reply]
I am having the exact same issue. Were you able to fix it?

**Reply by syberphunk** [reply]
No

**Comment by abhishek-ch** [comment]
When I am trying https://huggingface.co/TheBloke/medicine-LLM-GGUF/blob/main/medicine-llm.Q8_0.gguf?download=true ,  its going for forever downloading without any progress beyond 0

**Comment by hemangjoshi37a** [comment]
now not even my ollama is getting connected with open-webui i did so much work but did not work

**Comment by Andreaux** [comment]
I'm having the exact same issue. I am absolutely unable to add through either upload or URL downloading any GGUF model file. Nothing works :(

**Comment by mysterium-coniunctionis** [comment]
Same errors as others here - unable to complete the GGUF upload. I am running two instances of Open WebUI + Ollama:

1) When attempting to "Upload a GGUF model" via my M1 MacBook Pro Ollama (official macOS app) + Docker Desktop installation of Open WebUI. GGUF files will upload to 100% and then they just hang forever. It used to be that there was a slight delay at this point but then the modelfile would be updated and the upload would complete. Now, just stuck. 

2) I also have Ollama (docker container with a dedicated NVIDIA GPU) + Open WebUI (Cloudron app) installed on my QNAP NAS and that one just stays stuck at 0% when it used to work typically much faster than the instance on my laptop.

**Comment by sivertheisholt** [comment]
Same error here, tried a few versions without any luck.

Edit: Removed all files + clean install fixed the problem for me. Not sure why because it's the exact same setup/settings.

**Comment by syberphunk** [comment]
> I'm experiencing this problem also, except it's saying that the connection was aborted when attempting to upload. The gguf file appears in the upload folder but doesn't get any further
> 
> ![image](https://private-user-images.githubusercontent.com/234857/327814703-caaffede-be27-45cd-a02c-587fc8fbaeaf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc0NTEzNzIsIm5iZiI6MTcxNzQ1MTA3MiwicGF0aCI6Ii8yMzQ4NTcvMzI3ODE0NzAzLWNhYWZmZWRlLWJlMjctNDVjZC1hMDJjLTU4N2ZjOGZiYWVhZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwM1QyMTQ0MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMzFjOGEzYWMxNzZkNDdiYjhjMTgwNDk5NWRlZDA4MDY1NjRiOTAxMWM4NjI1ZGExZDBiZmIxOGMxMzVhNjY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.61tQvUuD_xAuP4FCKoq96CYKM2waIgoAfafoiAPW2z0)

Unfortunately I still have this problem
![image](https://github.com/open-webui/open-webui/assets/234857/09d9ede3-2e70-4738-9b34-928e45720cdb)

**Comment by SchneiderSam** [comment]
@justinh-rahb i got this error:
![image](https://github.com/open-webui/open-webui/assets/9355398/03cf8dd3-6036-42c9-ab25-d63f9f08e999)

**Comment by tjbck** [comment]
GGUF file upload for Ollama will remain experimental (will not work for certain cases), file size exceeding 4gb are known to have issues with the upload process. I'll be moving this to discussion, however, If anyone's interested in fixing the issue, feel free to make a PR!

**Reply by syberphunk** [reply]
> however, If anyone's interested in fixing the issue, feel free to make a PR!

Sigh.

Do you at least have detail on why this happens to help steer someone into a solution?

**Comment by imrshohel** [comment]
I'm facing the same issue. It's never worked and never will. Downloading big files repeatedly and multiple versions is kind of a headache right now.

My C drives and my specific LLM-based drive are now showing red because I have to download multiple versions. The reason is that I cannot upload my model directly in Open WebUI. Every time, it gets stuck and never shows me that it's uploaded.

**Comment by Goo6i** [comment]
I tried uploading a 8gb GGUF and it worked fine. However, I haven't tried using URL link so I cannot comment on that at the moment.

I'm using Ollama on my local windows machine and open-webui as a docker container on my synology nas.
Will try the URL thingy later and report back.

Edit: Downloading an 8GB GGUF file using the URL Mode seems to be working fine as well.
Can't reproduce mentioned failures.

**Reply by syberphunk** [reply]
> I tried uploading a 8gb GGUF and it worked fine. However, I haven't tried using URL link so I cannot comment on that at the moment.
> 
> I'm using Ollama on my local windows machine and open-webui as a docker container on my synology nas. Will try the URL thingy later and report back.
> 
> Edit: Downloading an 8GB GGUF file using the URL Mode seems to be working fine as well. Can't reproduce mentioned failures.


Is your main windows account running as an administrator? Or do you need to elevate privileges to run things as an administrator?

I'm wondering if that's the main cause, nvidia's software hits that issue where it expects your user account to be in the administrators group - which just isn't good from a security perspective.

I still have the upload problem with this which is driving me insane.

**Reply by Goo6i** [reply]
Yeah that's correct I'm running as an admin. I'll try making a non-admin user and test the same conditions to see if anything changes.

**Reply by Goo6i** [reply]
Using a standard account also worked fine. I can't find any issue whatsoever with the URL Mode.
Are you using some sort of memory management setting that "puts your tabs to sleep"? this could be a possible issue.

**Reply by syberphunk** [reply]
Url mode isnt the problem, its file upload that's the problem.

**Reply by Goo6i** [reply]
I also tested that as well and it worked out fine. I failed to mention that earlier. Apologies.

**Reply by syberphunk** [reply]
I'm also not using memory management to put tabs to sleep.

There's not enough information here to do a clear comparison as to why you're able to do this. Perhaps if you could lay out your setup and run steps along with information on the browser and any extensions you have installed or any settings you may have changed.

**Comment by MattFromTheFuture** [comment]
I too was having issues uploading a GGUF model using both the URL and file method.  I usually access openweb ui via a reverse proxy, and that's how I've been trying upload the GGUF, but decided to try and upload direct via IP and I didn't' get any issues.  I probably have a misconfiguration in my reverse proxy but hopefully this helps someone
